snap/snapenv/snapenv.go

```diff
@@ -77,6 +79,6 @@ func GetBasicSnapEnvVars(desc interface{}) []string {
 // somewhere more reasonable like the snappy module.
 func GetUserSnapEnvVars(desc interface{}) []string {
 	return fillSnapEnvVars(desc, []string{
-		"SNAP_USER_DATA={{.Home}}{{.SnapPath}}",
+		"SNAP_USER_DATA={{.Home}}{{.App.Snap.MountDir}}",
```

Comment:
this is still a bit of a hack, as we discussed at some point using templates here is a bit overkill and tempt into doing strange things, anyway something that can be fixed in a later pass